### PR TITLE
Implementing throttling policies using siddhi extensions

### PR DESCRIPTION
**Pull request by RameshaKaru**

### CODE REVIEW

File: components/micro-gateway-cli/src/main/resources/templates/policy.mustache [hunk 1]
```diff
@@ -12,56 +12,47 @@ stream<gateway:GlobalThrottleStreamDTO> s{{name}}globalThrotCopy = gateway:globa
...
+                    group by s{{name}}eligibilityStream.throttleKey
+                    => (gateway:IntermediateStream[] counts) {
+                    foreach var c in counts{
+                    s{{name}}intermediateStream.publish(c);
+                    }
+                    }
+
+                    from s{{name}}intermediateStream
+                    select s{{name}}intermediateStream.throttleKey, s{{name}}intermediateStream.eventCount>= {{count}} as isThrottled,
                     s{{name}}intermediateStream.stopOnQuota, s{{name}}intermediateStream.expiryTimeStamp
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/resources/templates/policy.mustache]: Format the above code

File: components/micro-gateway-cli/src/main/resources/templates/policy.mustache [hunk 2]
```diff
@@ -12,56 +12,47 @@ stream<gateway:GlobalThrottleStreamDTO> s{{name}}globalThrotCopy = gateway:globa
...
+
+                    from s{{name}}resultStream
+                    throttler:emitOnStateChange(s{{name}}resultStream.throttleKey, s{{name}}resultStream.isThrottled)
+                    select s{{name}}resultStream.throttleKey as throttleKey, s{{name}}resultStream.isThrottled,
+                    s{{name}}resultStream.stopOnQuota, s{{name}}resultStream.expiryTimeStamp
+                    => (gateway:GlobalThrottleStreamDTO[] counts) {
+                    foreach var c in counts{
+                    s{{name}}globalThrotCopy.publish(c);
+                    }
+                    }
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/resources/templates/policy.mustache]: Format the above code

**Comment by Rajith90** [comment]
depends on PR : https://github.com/wso2/product-microgateway/pull/397

File: components/micro-gateway-cli/src/main/resources/policies/application_50PerMin.bal [hunk 3]
```diff
@@ -7,62 +7,52 @@ import wso2/gateway;
...
+        throttler:timeBatch(60000)
         where s50PerMineligibilityStream.isEligible == true
-        select s50PerMineligibilityStream.throttleKey as throttleKey, count() as eventCount, true as stopOnQuota,
-        s50PerMineligibilityStream.expiryTimestamp as expiryTimeStamp
+        select s50PerMineligibilityStream.throttleKey as throttleKey, count() as eventCount, true as
+        stopOnQuota, expiryTimeStamp
         group by s50PerMineligibilityStream.throttleKey
         => (gateway:IntermediateStream[] counts) {
-            foreach var c in counts {
+            foreach var c in counts{
```
Comment (hasuniea) [inline components/micro-gateway-cli/src/main/resources/policies/application_50PerMin.bal]: Fix the formatting issue.

File: components/micro-gateway-cli/src/main/resources/policies/application_10PerMin.bal [hunk 4]
```diff
@@ -7,61 +7,52 @@ import wso2/gateway;
 stream<gateway:IntermediateStream> s10PerMinintermediateStream = new;
 stream<gateway:GlobalThrottleStreamDTO> s10PerMinresultStream = new;
 stream<gateway:EligibilityStreamDTO> s10PerMineligibilityStream = new;
-stream<gateway:RequestStreamDTO> s10PerMinreqCopy = gateway:requestStream;
+stream<gateway:RequestStreamDTO> s10PerMinreqCopy= gateway:requestStream;
```
Comment (hasuniea) [inline components/micro-gateway-cli/src/main/resources/policies/application_10PerMin.bal]: fix here as well.

File: components/micro-gateway-cli/src/main/resources/policies/application_50PerMin.bal (hunk 3 above)
Comment (RameshaKaru) [reply components/micro-gateway-cli/src/main/resources/policies/application_50PerMin.bal]: formatted

File: components/micro-gateway-cli/src/main/resources/policies/application_10PerMin.bal (hunk 4 above)
Comment (RameshaKaru) [reply components/micro-gateway-cli/src/main/resources/policies/application_10PerMin.bal]: formatted

**Pull request merged**